### PR TITLE
UserRedux updates for login and favoriting

### PR DESCRIPTION
**Pull request by ekayy**

### CODE REVIEW

File: Containers/FitDetail.js [hunk 1]
```diff
@@ -54,44 +54,24 @@ class FitDetail extends Component {
...
-        this.setState({ toggled: true });
-      } catch (error) {
-        console.tron.log(error);
-      }
-    }
+    await this.props.favoriteFit(id, this.props.user);
+
+    this.getFavoriteState();
+
+    this.getFavoriteState();
```
Comment (wesleylulee) [inline Containers/FitDetail.js:74]: why does this happen twice?

File: Containers/Profile.js [hunk 2]
```diff
@@ -151,7 +192,8 @@ const styles = {
 
 const mapStateToProps = state => {
   return {
-    favorites: state.user.favorites,
+    favorite_garments: state.user.favorite_garments,
```
Comment (wesleylulee) [inline Containers/Profile.js:195]: sometimes we use favorite_garments an sometimes we use favoriteGarments. Aside from favorite_garments coming from the backend, does the front end ever use it?

File: Containers/FitDetail.js (hunk 1 above)
Comment (ekayy) [reply Containers/FitDetail.js:74]: Good catch! Will fix

File: Containers/Profile.js (hunk 2 above)
Comment (ekayy) [reply Containers/Profile.js:195]: favorite_garments comes from the backend response which is being stored using redux. Calling this.props.favorite_garments is pulling from that store. I kept the name initially, to be consistent with the api. One of the benefits is being able to transfer and access application state easily from different components.

this.props.favorite_garments is different from this.state.favoriteGarments. One is coming from the redux store, and the other is an internal component level state.

I can definitely map to camelCase variables in UserRedux if you feel that is cleaner.

**Review by wesleylulee — APPROVED** [review]

**Pull request merged**